diff --git a/u-boot/board/ti/am335x/board.c b/u-boot/board/ti/am335x/board.c
index 34f987c2b7..b47018335e 100644
--- a/u-boot/board/ti/am335x/board.c
+++ b/u-boot/board/ti/am335x/board.c
@@ -949,6 +949,15 @@ U_BOOT_DRVINFO(am335x_eth) = {
 #endif
 
 #ifdef CONFIG_SPL_LOAD_FIT
+int oresat_name_compare(const char *name, const char *name2) {
+	char *rev = board_ti_get_rev();
+	ssize_t name2_len = strlen(name2);
+	char buffer[name2_len  + 4];
+	strncpy(buffer, name2, name2_len);
+	strncpy(&buffer[name2_len], rev, 4);
+	return !strncmp(name, buffer, name2_len + 4);
+}
+
 int board_fit_config_name_match(const char *name)
 {
 	if (board_is_gp_evm() && !strcmp(name, "am335x-evm"))
@@ -978,6 +987,16 @@ int board_fit_config_name_match(const char *name)
 			return 0;
 		}
 	}
+	else if (board_is_oresat_c3() && oresat_name_compare(name, "oresat-c3-"))
+		return 0;
+	else if (board_is_oresat_gps() && oresat_name_compare(name, "oresat-gps-"))
+		return 0;
+	else if (board_is_oresat_st() && oresat_name_compare(name, "oresat-star-tracker-"))
+		return 0;
+	else if (board_is_oresat_dxwifi() && oresat_name_compare(name, "oresat-dxwifi-"))
+		return 0;
+	else if (board_is_oresat_cfc() && oresat_name_compare(name, "oresat-cfc-"))
+		return 0;
 	return -1;
 }
 #endif
diff --git a/u-boot/board/ti/am335x/board.h b/u-boot/board/ti/am335x/board.h
index 1284c160d8..ee18e2bd94 100644
--- a/u-boot/board/ti/am335x/board.h
+++ b/u-boot/board/ti/am335x/board.h
@@ -80,6 +80,26 @@ static inline int board_is_icev2(void)
 	return board_ti_is("A335_ICE") && !strncmp("2", board_ti_get_rev(), 1);
 }
 
+static inline int board_is_oresat_c3(void) {
+	return board_ti_is("A335OSC3");
+}
+
+static inline int board_is_oresat_gps(void) {
+	return board_ti_is("A335OGPS");
+}
+
+static inline int board_is_oresat_st(void) {
+	return board_ti_is("A335OSST");
+}
+
+static inline int board_is_oresat_dxwifi(void) {
+	return board_ti_is("A335ODWF");
+}
+
+static inline int board_is_oresat_cfc(void) {
+	return board_ti_is("A335OCFC");
+}
+
 /*
  * We have three pin mux functions that must exist.  We must be able to enable
  * uart0, for initial output and i2c0 to read the main EEPROM.  We then have a
diff --git a/u-boot/include/configs/am335x_evm.h b/u-boot/include/configs/am335x_evm.h
index 67f4656253..61acbde465 100644
--- a/u-boot/include/configs/am335x_evm.h
+++ b/u-boot/include/configs/am335x_evm.h
@@ -138,6 +138,16 @@
 			"setenv fdtfile am335x-evm.dtb; fi; " \
 		"if test $board_name = A335X_SK; then " \
 			"setenv fdtfile am335x-evmsk.dtb; fi; " \
+		"if test $board_name = A335OSC3; then " \
+			"setenv fdtfile oresat-c3-${board_rev}.dtb; fi; " \
+		"if test $board_name = A335OGPS; then " \
+			"setenv fdtfile oresat-gps-${board_rev}.dtb; fi; " \
+		"if test $board_name = A335OSST; then " \
+			"setenv fdtfile oresat-star-tracker-${board_rev}.dtb; fi; " \
+		"if test $board_name = A335ODWF; then " \
+			"setenv fdtfile oresat-dxwifi-${board_rev}.dtb; fi; " \
+		"if test $board_name = A335OCFC; then " \
+			"setenv fdtfile oresat-cfc-${board_rev}.dtb; fi; " \
 		"if test $board_name = A335_ICE; then " \
 			"setenv fdtfile am335x-icev2.dtb; " \
 			"if test $ice_mii = mii; then " \
